### PR TITLE
2015 11 mn move column functions to directive+1

### PR DESCRIPTION
**Pull request by local-girl**
Please note this is work in progress. It basically works in Mercator, I will add functionality for other sites as I come across it rather than pre-empting functionality.

Here are places we need to put the action bar
# Mercator
- [x] proposal detail column
- [ ] proposal list column (filter and sort) Functionality not available in this version of action bar
- [x] new proposal (cancel only)
- [x] new proposal (cancel only)

# SPD
- [ ] list documents (create document)
- [ ] Document detail (edit document)

# MeinBerlin
## kiezkasse
- [ ] List documents (mapSwitcher)
- [ ] Proposal detail (edit)
- [ ] Proposal edit (cancel)
- [ ] MISSING FROM CURRENT? Proposal create (cancel)

## Alex
- [ ] List documents (mapSwitcher)
- [ ] List documents / document tab (create document)
- [ ] Create Proposal (cancel)
- [ ] Proposal Detail (edit)
- [ ] Proposal Edit (cancel)
- [ ] Create Document (cancel)
- [ ] Document Detail (edit, cancel)
- [ ] Document Edit (cancel)

## BPlan
?? Can't run it ?

## Stadtforum
?? Can't run it ?

## Buergerhaushalt
- [ ] List Proposals (Map switch)
- [ ] MISSING Create Proposal (cancel)
- [ ] Proposal Detail (edit)
- [ ] Proposal Edit (cancel)

# Policy Compass
- [ ] Process Detail (create)
- [ ] Proposal Detail (edit, create badge assignments (does not exist in actionbar!!))
- [ ] Proposal Edit (cancel)

### CODE REVIEW

**Comment by xi** [comment]
Can #1807 be closed now?

**Comment by local-girl** [comment]
Yes

**Comment by local-girl** [comment]
OK this is ready for a merge, It just needs to be implemented and incorporate things such as the map switcher. I'll add these things in a further branch but this is getting too large now.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Facets.html [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
+                href=""
+                data-ng-repeat="item in facet.items"
+                class="facet-item-{{item.key}} facet-item button-cta-secondary"
+                data-ng-class="{'is-active': item.enabled}"
+                data-ng-click="toggleItem(facet, item, $event)">
+                {{ item.name | translate }}
+            </a>
+        </div>
+    </div>
+</div>
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Facets.html]: Why did you change this? IMHO it has less symantics and no real advantages. See also  5fc7910.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"delete();\">{{ \"TR__DELETE\" | translate }}</a>",
+        require: "^adhMovingColumn",
+        scope: {
+            resourcePath: "@",
+            class: "@"
+        },
+        link: (scope, element, attrs, column : AdhMovingColumns.MovingColumnController) => {
+            scope.delete = () => {
+                column.$broadcast("triggerDelete", scope.proposalUrl);
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: THis only works in one very specific case (mercator proposal).

**Comment by xi** [comment]
@local-girl can you please provide a list of columns that need to be adapted?

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        require: "^adhMovingColumn",
+        scope: {
+            resourcePath: "@",
+            class: "@"
+        },
+        link: (scope) => {
+            scope.edit = () => {
+                var url = adhResourceUrl(scope.resourcePath, "edit");
+                $location.url(url);
+            };
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: wouldn't it be better to use a simple link here?

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"print();\">{{ \"TR__PRINT\" | translate }}</a>",
+        scope: {
+            class: "@"
+        },
+        link: () => {
+            scope.print = () => {
+                // only the focused column is printed
+                adhTopLevelState.set("focus", 1);
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: The `1` here should be replaced by the index of the current column.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    adhResourceUrl,
+    $location : angular.ILocationService
+) => {
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"edit();\">{{ \"TR__EDIT\" | translate }}</a>",
+        require: "^adhMovingColumn",
+        scope: {
+            resourcePath: "@",
+            class: "@"
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: shouldn't this also have the `parentPath` option?

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Facets.html (hunk 1 above)
Comment (local-girl) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Facets.html]: I don't think a list suits this group of buttons, thought this would be lighter

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 2 above)
Comment (local-girl) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: Would be a shame if we didn't standardise stuff like this, otherwise we have to just have a pass link function

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 3 above)
Comment (local-girl) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: Again, not if we can standardise edit

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 4 above)
Comment (local-girl) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: Thats what was there before I believe

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 5 above)
Comment (local-girl) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: Yes I was just doing these as per needed, was going to add this in future revisions.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 4 above)
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: True, but before this code was only used in the second column.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 3 above)
Comment (local-girl) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: My plan was to create a standard edit function here, I'll probably need some guidance.

**Comment by xi** [comment]
Nice work!
